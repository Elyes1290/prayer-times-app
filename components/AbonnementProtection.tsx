import React, { useEffect, useRef } from "react";
import { BackHandler, Alert } from "react-native";
import AsyncStorage from "@react-native-async-storage/async-storage";

interface AbonnementProtectionProps {
  children: React.ReactNode;
}

/**
 * üõ°Ô∏è Composant de protection contre les redirections automatiques
 * pendant la s√©lection d'abonnement
 */
const AbonnementProtection: React.FC<AbonnementProtectionProps> = ({
  children,
}) => {
  const protectionRef = useRef(false);
  const backHandlerRef = useRef<any>(null);

  useEffect(() => {
    const checkAbonnementProcess = async () => {
      try {
        const pendingRegistration = await AsyncStorage.getItem(
          "pending_registration"
        );
        if (pendingRegistration) {
          if (!protectionRef.current) {
            protectionRef.current = true;
            console.log("üõ°Ô∏è Protection d'abonnement activ√©e");

            // üö® NOUVEAU : Bloquer le bouton retour pendant l'abonnement
            backHandlerRef.current = BackHandler.addEventListener(
              "hardwareBackPress",
              () => {
                Alert.alert(
                  "Abonnement en cours",
                  "Veuillez terminer votre abonnement avant de quitter cette page.",
                  [{ text: "OK" }]
                );
                return true; // Bloquer le retour
              }
            );
          }
        } else {
          if (protectionRef.current) {
            protectionRef.current = false;
            console.log("üõ°Ô∏è Protection d'abonnement d√©sactiv√©e");

            // D√©sactiver le blocage du bouton retour
            if (backHandlerRef.current) {
              backHandlerRef.current.remove();
              backHandlerRef.current = null;
            }
          }
        }
      } catch (error) {
        console.error("‚ùå Erreur v√©rification protection abonnement:", error);
      }
    };

    // V√©rifier imm√©diatement
    checkAbonnementProcess();

    // V√©rifier toutes les 2 secondes
    const interval = setInterval(checkAbonnementProcess, 2000);

    return () => {
      clearInterval(interval);
      if (backHandlerRef.current) {
        backHandlerRef.current.remove();
      }
    };
  }, []);

  return <>{children}</>;
};

export default AbonnementProtection;
