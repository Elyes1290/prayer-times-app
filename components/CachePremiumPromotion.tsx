// components/CachePremiumPromotion.tsx
import React from "react";
import { View, Text, TouchableOpacity, StyleSheet } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { useThemeColors } from "../hooks/useThemeAssets";
import { useCurrentTheme } from "../hooks/useThemeColor";

interface CachePremiumPromotionProps {
  onUpgrade: () => void;
  visible?: boolean;
}

export default function CachePremiumPromotion({
  onUpgrade,
  visible = true,
}: CachePremiumPromotionProps) {
  const colors = useThemeColors();
  const currentTheme = useCurrentTheme();

  if (!visible) return null;

  const styles = StyleSheet.create({
    container: {
      backgroundColor:
        currentTheme === "light" ? colors.cardBG : "rgba(0, 0, 0, 0.8)",
      borderRadius: 12,
      padding: 16,
      marginVertical: 8,
      borderWidth: 1,
      borderColor:
        currentTheme === "light" ? colors.border : "rgba(255, 255, 255, 0.1)",
    },
    header: {
      flexDirection: "row",
      alignItems: "center",
      marginBottom: 12,
    },
    icon: {
      marginRight: 8,
    },
    title: {
      fontSize: 16,
      fontWeight: "600",
      color: colors.text,
      flex: 1,
    },
    premiumBadge: {
      backgroundColor: "#FFD700",
      paddingHorizontal: 8,
      paddingVertical: 4,
      borderRadius: 12,
    },
    premiumText: {
      fontSize: 12,
      fontWeight: "700",
      color: "#000",
    },
    description: {
      fontSize: 14,
      color: colors.textSecondary,
      lineHeight: 20,
      marginBottom: 12,
    },
    features: {
      marginBottom: 16,
    },
    feature: {
      flexDirection: "row",
      alignItems: "center",
      marginBottom: 8,
    },
    featureIcon: {
      marginRight: 8,
    },
    featureText: {
      fontSize: 13,
      color: colors.textSecondary,
      flex: 1,
    },
    upgradeButton: {
      backgroundColor: colors.primary,
      paddingVertical: 12,
      paddingHorizontal: 16,
      borderRadius: 8,
      flexDirection: "row",
      alignItems: "center",
      justifyContent: "center",
    },
    upgradeButtonText: {
      color: "#FFFFFF",
      fontSize: 14,
      fontWeight: "600",
      marginLeft: 8,
    },
  });

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <MaterialCommunityIcons
          name="lightning-bolt"
          size={20}
          color="#FFD700"
          style={styles.icon}
        />
        <Text style={styles.title}>Cache Premium</Text>
        <View style={styles.premiumBadge}>
          <Text style={styles.premiumText}>PREMIUM</Text>
        </View>
      </View>

      <Text style={styles.description}>
        Débloquez le cache avancé pour une expérience optimale hors ligne
      </Text>

      <View style={styles.features}>
        <View style={styles.feature}>
          <MaterialCommunityIcons
            name="calendar-month"
            size={16}
            color="#4CAF50"
            style={styles.featureIcon}
          />
          <Text style={styles.featureText}>
            Préchargement 30 jours (vs 7 jours gratuit)
          </Text>
        </View>

        <View style={styles.feature}>
          <MaterialCommunityIcons
            name="download"
            size={16}
            color="#4CAF50"
            style={styles.featureIcon}
          />
          <Text style={styles.featureText}>
            Préchargement automatique en arrière-plan
          </Text>
        </View>

        <View style={styles.feature}>
          <MaterialCommunityIcons
            name="speedometer"
            size={16}
            color="#4CAF50"
            style={styles.featureIcon}
          />
          <Text style={styles.featureText}>
            Calculs simultanés optimisés (5x plus rapide)
          </Text>
        </View>

        <View style={styles.feature}>
          <MaterialCommunityIcons
            name="cloud-sync"
            size={16}
            color="#4CAF50"
            style={styles.featureIcon}
          />
          <Text style={styles.featureText}>
            Synchronisation cloud des préférences
          </Text>
        </View>
      </View>

      <TouchableOpacity style={styles.upgradeButton} onPress={onUpgrade}>
        <MaterialCommunityIcons name="crown" size={20} color="#FFFFFF" />
        <Text style={styles.upgradeButtonText}>Passer au Premium</Text>
      </TouchableOpacity>
    </View>
  );
}
