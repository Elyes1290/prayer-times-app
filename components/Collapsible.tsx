import Ionicons from "@expo/vector-icons/Ionicons";
import { PropsWithChildren, useState } from "react";
import { StyleSheet, TouchableOpacity } from "react-native";

import { ThemedText } from "@/components/ThemedText";
import { ThemedView } from "@/components/ThemedView";
import { useThemeColor } from "@/hooks/useThemeColor";

export function Collapsible({
  children,
  title,
}: PropsWithChildren & { title: string }) {
  const [isOpen, setIsOpen] = useState(false);
  const iconColor = useThemeColor({}, "text");

  return (
    <ThemedView testID="collapsible-container">
      <TouchableOpacity
        style={styles.heading}
        onPress={() => setIsOpen((value) => !value)}
        activeOpacity={0.8}
        testID="collapsible-header"
      >
        <Ionicons
          name={isOpen ? "chevron-down" : "chevron-forward-outline"}
          size={18}
          weight="medium"
          color={iconColor}
          testID="collapsible-icon"
        />
        <ThemedText type="defaultSemiBold" testID="collapsible-title">
          {title}
        </ThemedText>
      </TouchableOpacity>
      {isOpen && (
        <ThemedView style={styles.content} testID="collapsible-content">
          {children}
        </ThemedView>
      )}
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  heading: {
    flexDirection: "row",
    alignItems: "center",
    gap: 6,
  },
  content: {
    marginTop: 6,
    marginLeft: 24,
  },
});
