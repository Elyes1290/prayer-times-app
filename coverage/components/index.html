
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> components</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">44.13% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>256/580</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">40.27% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>232/576</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">44.44% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>56/126</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">45.17% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>248/549</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="AbonnementProtection.tsx"><a href="AbonnementProtection.tsx.html">AbonnementProtection.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="28" class="abs low">0/28</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="28" class="abs low">0/28</td>
	</tr>

<tr>
	<td class="file low" data-value="BadgesManager.tsx"><a href="BadgesManager.tsx.html">BadgesManager.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="68" class="abs low">0/68</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="54" class="abs low">0/54</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="65" class="abs low">0/65</td>
	</tr>

<tr>
	<td class="file high" data-value="Collapsible.tsx"><a href="Collapsible.tsx.html">Collapsible.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file high" data-value="CollapsibleSection.tsx"><a href="CollapsibleSection.tsx.html">CollapsibleSection.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="68.42" class="pct medium">68.42%</td>
	<td data-value="19" class="abs medium">13/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

<tr>
	<td class="file low" data-value="DateNavigator.tsx"><a href="DateNavigator.tsx.html">DateNavigator.tsx</a></td>
	<td data-value="31.42" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 31%"></div><div class="cover-empty" style="width: 69%"></div></div>
	</td>
	<td data-value="31.42" class="pct low">31.42%</td>
	<td data-value="35" class="abs low">11/35</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="40" class="abs low">16/40</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="45.45" class="pct low">45.45%</td>
	<td data-value="22" class="abs low">10/22</td>
	</tr>

<tr>
	<td class="file high" data-value="ExternalLink.tsx"><a href="ExternalLink.tsx.html">ExternalLink.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="FavoriteButton.tsx"><a href="FavoriteButton.tsx.html">FavoriteButton.tsx</a></td>
	<td data-value="80.64" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 80%"></div><div class="cover-empty" style="width: 20%"></div></div>
	</td>
	<td data-value="80.64" class="pct high">80.64%</td>
	<td data-value="62" class="abs high">50/62</td>
	<td data-value="45.76" class="pct low">45.76%</td>
	<td data-value="59" class="abs low">27/59</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="81.96" class="pct high">81.96%</td>
	<td data-value="61" class="abs high">50/61</td>
	</tr>

<tr>
	<td class="file medium" data-value="HapticTab.tsx"><a href="HapticTab.tsx.html">HapticTab.tsx</a></td>
	<td data-value="75" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 75%"></div><div class="cover-empty" style="width: 25%"></div></div>
	</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file low" data-value="HelloWave.tsx"><a href="HelloWave.tsx.html">HelloWave.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	</tr>

<tr>
	<td class="file low" data-value="NativeDownloadTest.tsx"><a href="NativeDownloadTest.tsx.html">NativeDownloadTest.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	</tr>

<tr>
	<td class="file low" data-value="ParallaxScrollView.tsx"><a href="ParallaxScrollView.tsx.html">ParallaxScrollView.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	</tr>

<tr>
	<td class="file low" data-value="PaymentDebugInfo.tsx"><a href="PaymentDebugInfo.tsx.html">PaymentDebugInfo.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="31" class="abs low">0/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="28" class="abs low">0/28</td>
	</tr>

<tr>
	<td class="file low" data-value="PaymentSyncDiagnostic.tsx"><a href="PaymentSyncDiagnostic.tsx.html">PaymentSyncDiagnostic.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="31" class="abs low">0/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="20" class="abs low">0/20</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="29" class="abs low">0/29</td>
	</tr>

<tr>
	<td class="file high" data-value="PrayerStats.tsx"><a href="PrayerStats.tsx.html">PrayerStats.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="26" class="abs medium">13/26</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	</tr>

<tr>
	<td class="file low" data-value="PremiumPromotion.tsx"><a href="PremiumPromotion.tsx.html">PremiumPromotion.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="25" class="abs low">0/25</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="21" class="abs low">0/21</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="25" class="abs low">0/25</td>
	</tr>

<tr>
	<td class="file low" data-value="RateLimitDiagnostic.tsx"><a href="RateLimitDiagnostic.tsx.html">RateLimitDiagnostic.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="42" class="abs low">0/42</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="42" class="abs low">0/42</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="42" class="abs low">0/42</td>
	</tr>

<tr>
	<td class="file high" data-value="SunInfo.tsx"><a href="SunInfo.tsx.html">SunInfo.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="51" class="abs high">51/51</td>
	<td data-value="76.92" class="pct medium">76.92%</td>
	<td data-value="52" class="abs medium">40/52</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="46" class="abs high">46/46</td>
	</tr>

<tr>
	<td class="file high" data-value="ThemedAlert.tsx"><a href="ThemedAlert.tsx.html">ThemedAlert.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="40" class="abs high">40/40</td>
	<td data-value="98.24" class="pct high">98.24%</td>
	<td data-value="57" class="abs high">56/57</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="40" class="abs high">40/40</td>
	</tr>

<tr>
	<td class="file high" data-value="ThemedImageBackground.tsx"><a href="ThemedImageBackground.tsx.html">ThemedImageBackground.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file low" data-value="ThemedPicker.tsx"><a href="ThemedPicker.tsx.html">ThemedPicker.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	</tr>

<tr>
	<td class="file high" data-value="ThemedText.tsx"><a href="ThemedText.tsx.html">ThemedText.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="ThemedView.tsx"><a href="ThemedView.tsx.html">ThemedView.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file high" data-value="Toast.tsx"><a href="Toast.tsx.html">Toast.tsx</a></td>
	<td data-value="96.15" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.15" class="pct high">96.15%</td>
	<td data-value="26" class="abs high">25/26</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="10" class="abs high">9/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="24" class="abs high">24/24</td>
	</tr>

<tr>
	<td class="file high" data-value="WeeklyPrayerView.tsx"><a href="WeeklyPrayerView.tsx.html">WeeklyPrayerView.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	<td data-value="76.59" class="pct medium">76.59%</td>
	<td data-value="47" class="abs medium">36/47</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	</tr>

<tr>
	<td class="file medium" data-value="WelcomePersonalizationModal.tsx"><a href="WelcomePersonalizationModal.tsx.html">WelcomePersonalizationModal.tsx</a></td>
	<td data-value="63.63" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 63%"></div><div class="cover-empty" style="width: 37%"></div></div>
	</td>
	<td data-value="63.63" class="pct medium">63.63%</td>
	<td data-value="22" class="abs medium">14/22</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="42.85" class="pct low">42.85%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="63.63" class="pct medium">63.63%</td>
	<td data-value="22" class="abs medium">14/22</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2025-08-20T15:38:48.371Z
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    