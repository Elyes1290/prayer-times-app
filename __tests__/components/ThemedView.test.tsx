import React from "react";
import { render } from "@testing-library/react-native";
import { ThemedView } from "../../components/ThemedView";

// Mock du hook useThemeColor
jest.mock("../../hooks/useThemeColor", () => ({
  useThemeColor: jest.fn(() => "#FFFFFF"),
}));

describe("ThemedView", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly with default props", () => {
    const { toJSON } = render(<ThemedView />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders with children", () => {
    const { getByTestId } = render(
      <ThemedView>
        <ThemedView testID="child" />
      </ThemedView>
    );

    expect(getByTestId("child")).toBeTruthy();
  });

  it("applies custom style", () => {
    const customStyle = { padding: 16 };
    const { getByTestId } = render(
      <ThemedView testID="themed-view" style={customStyle} />
    );

    const view = getByTestId("themed-view");
    expect(view.props.style).toContainEqual(customStyle);
  });

  it("passes through other props", () => {
    const { getByTestId } = render(
      <ThemedView testID="test-view" accessibilityLabel="Test Label" />
    );

    const view = getByTestId("test-view");
    expect(view.props.accessibilityLabel).toBe("Test Label");
  });

  it("uses light and dark colors", () => {
    const { useThemeColor } = require("../../hooks/useThemeColor");

    render(<ThemedView lightColor="#FF0000" darkColor="#00FF00" />);

    expect(useThemeColor).toHaveBeenCalledWith(
      { light: "#FF0000", dark: "#00FF00" },
      "background"
    );
  });

  it("handles multiple children", () => {
    const { getByTestId } = render(
      <ThemedView testID="parent">
        <ThemedView testID="child1" />
        <ThemedView testID="child2" />
      </ThemedView>
    );

    expect(getByTestId("child1")).toBeTruthy();
    expect(getByTestId("child2")).toBeTruthy();
  });
});
