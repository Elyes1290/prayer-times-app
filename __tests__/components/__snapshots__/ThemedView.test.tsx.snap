// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedView renders correctly with default props 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      undefined,
    ]
  }
/>
`;
